### ISSUE TITLE
DB Error 1064 after updating to 1.14.1 from 1.11

### ISSUE THREAD

**Issue by dszego** [osTicket/osTicket#5304]
Very similar to  #4795 - We did a fresh install of 1.14.1, then mysql-dumped in an old database from 1.11 ...

On nearly every transaction, we are now getting this error, similar to issue #4795 :

```
[SELECT COUNT(DISTINCT CASE WHEN A1.`est_duedate` < '2017-08-28 00:00:00' THEN A1.`ticket_id` END) AS `q15`, COUNT(DISTINCT CASE WHEN A1.`staff_id` IN (1) THEN A1.`ticket_id` END) AS `q16`, COUNT(DISTINCT CASE WHEN THEN A1.`ticket_id` END) AS `q20`, COUNT(DISTINCT CASE WHEN THEN A1.`ticket_id` END) AS `q30`, COUNT(DISTINCT CASE WHEN THEN A1.`ticket_id` END) AS `q32`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q1`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A2.`state` = 'open' AND A1.`isanswered` = '0' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q2`, COUNT(DISTINCT CASE WHEN (A1.`team_id` IN (7, 9) OR A1.`staff_id` IN (1)) AND A2.`state` = 'open' AND (A1.`team_id` IN (7, 9) OR A1.`staff_id` IN (1)) AND A2.`state` = 'open' AND A1.`staff_id` IN (1) AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q6`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2019-12-20 05:00:00' AND '2019-12-21 04:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q9`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A2.`state` = 'open' AND A1.`isanswered` = '1' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q3`, COUNT(DISTINCT CASE WHEN (A1.`team_id` IN (7, 9) OR A1.`staff_id` IN (1)) AND A2.`state` = 'open' AND (A1.`team_id` IN (7, 9) OR A1.`staff_id` IN (1)) AND A2.`state` = 'open' AND NOT A1.`staff_id` IN (1) AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q7`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2019-12-19 05:00:00' AND '2019-12-20 04:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q10`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A2.`state` = 'open' AND A1.`isoverdue` = '1' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q4`, COUNT(DISTINCT CASE WHEN (A1.`team_id` IN (7, 9) OR A1.`staff_id` IN (1)) AND A2.`state` = 'open' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q5`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2019-12-16 05:00:00' AND '2019-12-23 04:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q11`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q8`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2019-12-01 05:00:00' AND '2020-01-01 04:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q12`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2019-10-01 04:00:00' AND '2020-01-01 04:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q13`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2019-01-01 05:00:00' AND '2020-01-01 04:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q14` FROM `ost_ticket` A1 JOIN `ost_ticket_status` A2 ON (A1.`status_id` = A2.`id`) LEFT JOIN `ost_thread` A3 ON (A3.`object_type` = 'T' AND A1.`ticket_id` = A3.`object_id`) LEFT JOIN `ost_thread_referral` A4 ON (A3.`id` = A4.`thread_id`) LEFT JOIN `ost_staff` A5 ON (A4.`object_type` = 'S' AND A4.`object_id` = A5.`staff_id`) LEFT JOIN `ost_thread` A6 ON (A1.`ticket_id` = A6.`object_id` AND A6.`object_type` = 'C') LEFT JOIN `ost_thread_referral` A7 ON (A6.`id` = A7.`thread_id`) LEFT JOIN `ost_staff` A8 ON (A7.`object_type` = 'S' AND A7.`object_id` = A8.`staff_id`) LEFT JOIN `ost_team` B0 ON (A4.`object_type` = 'E' AND A4.`object_id` = B0.`team_id`) LEFT JOIN `ost_team` B1 ON (A7.`object_type` = 'E' AND A7.`object_id` = B1.`team_id`) LEFT JOIN `ost_department` B2 ON (A4.`object_type` = 'D' AND A4.`object_id` = B2.`id`) LEFT JOIN `ost_department` B3 ON (A7.`object_type` = 'D' AND A7.`object_id` = B3.`id`) WHERE ((A2.`state` = 'open' AND (A1.`staff_id` = 1 OR A5.`staff_id` = 1 OR A6.`object_type` = 'C' AND A8.`staff_id` = 1 OR A1.`team_id` IN (7, 9) OR B0.`team_id` IN (7, 9) OR A6.`object_type` = 'C' AND B1.`team_id` IN (7, 9))) OR (A2.`state` IN ('open', 'closed') AND (A1.`dept_id` IN (1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21) OR B2.`id` IN (1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21))) OR A6.`object_type` = 'C' AND B3.`id` IN (1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21))]

You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'THEN A1.`ticket_id` END) AS `q20`, COUNT(DISTINCT CASE WHEN THEN A1.`ticket_id`' at line 1<br />
<br />
---- Backtrace ----<br />
#0 (root)/include/mysqli.php(200): osTicket->logDBError('DB Error #1064', '[SELECT COUNT(D...')<br />
#1 (root)/include/class.orm.php(3459): db_query('SELECT COUNT(DI...', true, true)<br />
#2 (root)/include/class.orm.php(3506): MySqlExecutor->execute()<br />
#3 (root)/include/class.orm.php(2093): MySqlExecutor->getArray()<br />
#4 (root)/include/class.orm.php(2043): HashArrayIterator->{closure}()<br />
#5 (root)/include/class.orm.php(2022): CallbackSimpleIterator->next()<br />
#6 (root)/include/class.orm.php(2031): CallbackSimpleIterator->rewind()<br />
#7 (root)/include/class.orm.php(1703): CallbackSimpleIterator->valid()<br />
#8 (root)/include/class.orm.php(1713): CachedResultSet->fillTo(9223372036854775807)<br />
#9 (root)/include/class.orm.php(1336): CachedResultSet->asArray()<br />
#10 (root)/include/class.orm.php(1359): QuerySet->all()<br />
#11 (root)/include/class.search.php(973): QuerySet->one()<br />
#12 (root)/scp/autocron.php(43): SavedQueue::counts(Object(StaffSession), false)<br />
#13 {main}
```
The problem being THEN A1.`ticket_id` END) AS `q20`, COUNT(DISTINCT CASE WHEN THEN A1.`ticket_id`, where **q20** =  **COUNT(DISTINCT CASE WHEN THEN A1.`ticket_id` END)** 
(I'm not sure what that means, so don't know how to find it and correct it in the database).

Table ost_queue, ID 20 is:

`{"keywords":"pedigree","status_id+search":"1","status_id+includes":"{\"1\":\"Open\"}"}`

The other ticket I referenced had a fit - a bit of database corruption that was manually corrected - but it's so specific to his tables/entries that I can't decypher it to match mine. See: [https://github.com/osTicket/osTicket/issues/4795#issuecomment-473935602](https://github.com/osTicket/osTicket/issues/4795#issuecomment-473935602)

Could someone kindly help walk me through finding the offending DB entry?

Thanks,
David.

**Comment by JediKev** [comment]
@dszego

Go to `ost_queue` table, copy the record with `id` of `20`, paste it here so we can do some research, and then delete the record in your database.

Cheers.

**Comment by dszego** [comment]
> Go to `ost_queue` table, copy the record with `id` of `20`, paste it here so we can do some research, and then delete the record in your database.

`{"keywords":"pedigree","status_id+search":"1","status_id+includes":"{\"1\":\"Open\"}"}`

Also added to the post. Thank you!

**Comment by glienhard** [comment]
I had the exact same issue when upgrading from v1.10 to 1.12

The solution that ultimately fixed the issue was to have each agent delete any old custom search they had and recreate it in the new version. This works out for the better because the search function in the older versions was not as good.

**Comment by DavidCasillasRivero** [comment]
@glienhard Thanks, that fix works. Hope there is a better workaround. Many people will probably have many searches saved and it seems not a viable option to everybody.

**Comment by JediKev** [comment]
@dszego @glienhard @DavidCasillasRivero 

Resolved with #5285. If the queue cannot produce counts or produces an error, we skip it and show `-`.

Cheers.
